packages/payload/src/duplicateDocument/index.ts

```diff
@@ -59,6 +62,11 @@ export const getDuplicateDocumentData = async ({
     req,
   })
 
+  if (selectedLocales && selectedLocales.length > 0 && duplicatedFromDocWithLocales) {
```

Comment:
I think this should actually loop over the collection fields and make sure that we are only trimming data on fields that are localized.

Reply:
> make sure that we are only trimming data on fields that are localized.

This is the current behavior, it is being handled in `filterLocales()` where we return any data that does not following a localized pattern i.e. `title: 'hello'` vs `title: {en: 'hello', es: 'hola'}`. I will add tests for this